app/src/main/cpp/skyline/services/am/storage/IStorageAccessor.cpp

```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MPL-2.0
+// Copyright © 2020 Skyline Team and Contributors (https://github.com/skyline-emu/)
+
+#include <kernel/types/KProcess.h>
+#include "IStorage.h"
+#include "IStorageAccessor.h"
+
+namespace skyline::service::am {
+    IStorageAccessor::IStorageAccessor(const DeviceState &state, ServiceManager &manager, std::shared_ptr<IStorage> parent) : parent(parent), BaseService(state, manager, Service::am_IStorageAccessor, "am:IStorageAccessor", {
+        {0x0, SFUNC(IStorageAccessor::GetSize)},
+        {0xb, SFUNC(IStorageAccessor::Read)}
+    }) {}
+
+    void IStorageAccessor::GetSize(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        response.Push<i64>(parent->content.size());
+    }
+
+    void IStorageAccessor::Read(type::KSession &session, ipc::IpcRequest &request, ipc::IpcResponse &response) {
+        auto offset = request.Pop<i64>();
+        auto size = request.outputBuf.at(0).size;
+
+        if (offset + size > parent->content.size())
+            throw exception("Trying to read past the end of an IStorage");
```

Comment:
Same as above

Reply:
Not sure what you mean

Reply:
["The supplied value cannot fit into the IStorage"](https://github.com/skyline-emu/skyline/pull/62#discussion_r450433823)

Reply:
That makes no sense in this context though

Reply:
I didn't read the function name properly, my bad. I noticed that `offset` is signed and negative values could cause completely OOB reads to unprotected data.